src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -984,7 +984,7 @@ private long GetAssemblyId(WriteObjectInfo objectInfo)
             {
                 assemId = 0;
             }
-            else if (assemblyString.Equals(Converter.s_urtAssemblyString))
+            else if (Converter.s_urtAssembly.GetType(objectInfo.GetTypeFullName(), throwOnError: false) == objectInfo._objectType)
```

Comment:
For reference, I just tried a little perf test serializing a very simple type:
```C#
[Serializable] class Foo { }
```
This line change causes it to serialize ~25-30% slower.  It would be great to find an alternative that doesn't incur such a regression.